source/Plugins.cpp

```diff
@@ -59,42 +59,50 @@ bool Plugin::IsValid() const
 
 
 
-// Try to load a plugin at the given path. Returns true if loading succeeded.
+// Attempt to load a plugin at the given path.
 const Plugin *Plugins::Load(const string &path)
 {
 	// Get the name of the folder containing the plugin.
 	size_t pos = path.rfind('/', path.length() - 2) + 1;
 	string name = path.substr(pos, path.length() - 1 - pos);
 
-	auto *plugin = plugins.Get(name);
-
 	string pluginFile = path + "plugin.txt";
+	string aboutText;
 
 	// Load plugin metadata from plugin.txt.
-	DataFile file(pluginFile);
-	for(const DataNode &child : file)
+	bool hasName = false;
+	for(const DataNode &child : DataFile(pluginFile))
 	{
 		if(child.Token(0) == "name" && child.Size() >= 2)
-			plugin->name = child.Token(1);
+		{
+			name = child.Token(1);
+			hasName = true;
+		}
 		else if(child.Token(0) == "about" && child.Size() >= 2)
-			plugin->aboutText = child.Token(1);
+			aboutText = child.Token(1);
+		else
+			child.PrintTrace("Skipping unrecognized attribute:");
 	}
 
-	// Set missing required values.
-	if(plugin->name.empty())
+	// 'name' is a required field for plugins with a plugin description file.
+	if(Files::Exists(pluginFile) && !hasName)
+		Logger::LogError("Warning: Missing required \"name\" field inside plugin.txt");
+
+	// Plugin names should be unique.
+	auto *plugin = plugins.Find(name);
+	if(plugin && plugin->IsValid())
 	{
-		plugin->name = std::move(name);
-		if(Files::Exists(pluginFile))
-		{
-			Logger::LogError(
-				"Failed to find name field in plugin.txt. Defaulting plugin name to folder name: \"" + plugin->name + "\"");
-		}
+		Logger::LogError("Warning: Skipping plugin located at \"" + path
+			+ "\" because another plugin with the same name has already been loaded");
+		return nullptr;
 	}
-	// Set values from old about.txt files.
-	if(plugin->aboutText.empty())
-		plugin->aboutText = Files::Read(path + "about.txt");
 
+	if(!plugin)
+		plugin = plugins.Get(name);
+	plugin->name = std::move(name);
```

Comment:
Why not just use "Get" in the first place?
Even if it does insert a new item into the collection, that will still return `false` to IsValid.